influxql/query_executor.go

```diff
@@ -349,44 +349,50 @@ func (e *QueryExecutor) query(qid uint64) (*QueryTask, bool) {
 //
 // After a query finishes running, the system is free to reuse a query id.
 func (e *QueryExecutor) attachQuery(q *Query, database string, interrupt <-chan struct{}) (uint64, *QueryTask, error) {
-	e.mu.Lock()
-	defer e.mu.Unlock()
+	qid, task, err := func() (uint64, *QueryTask, error) {
```

Comment:
I would suggest not making this an anonymous function, and manually unlock the mutex. `defer` allocates, and if this is already on a path of  contention, we'll want to unlock as soon as we know we can, rather than waiting for the return(357)-if(377)-return(378) path.